IPython/html/services/contents/filemanager.py

```diff
@@ -343,7 +367,7 @@ def get(self, path, content=True, type=None, format=None):
                 raise web.HTTPError(400,
                                 u'%s is a directory, not a %s' % (path, type), reason='bad type')
             model = self._dir_model(path, content=content)
-        elif type == 'notebook' or (type is None and path.endswith('.ipynb')):
+        elif type == 'notebook' or (type is None and path.endswith(self._nb_file_extensions)):
```

Comment:
I wonder if this might end up more maintainable if we added a `_is_notebook(path)` method that could be overridden in subclasses.  Then we don't have the (unused by default) extra complexity here, and other extensions are free to decide what "is this path a notebook means" in case they want to use something other than file extension to make that choice.